SecretManagement.KeePass.Extension/Tests/Test-SecretVault.Tests.ps1

```
@@ -76,20 +75,61 @@ InModuleScope -ModuleName 'SecretManagement.KeePass.Extension' {
         }
-        Context "Validating Path and UseMasterPassword Parameters" {
+        Context "Validating with incorrect MasterPassword" {
+            BeforeAll {
+                $MasterKey = 'You can not enter with this masterkey'
+                $VaultMasterKey = [PSCredential]::new('vaultkey', (ConvertTo-SecureString -AsPlainText -Force $MasterKey))
+
+                $VaultName = "KeepassPesterTest_$([guid]::NewGuid())"
+                $KeePassDatabaseSuffix = 'PathOnly'
+                $KeePassDatabaseFileName = "$($BaseKeepassDatabaseName)$($KeePassDatabaseSuffix).kdbx"
+                $VaultPath = Join-Path -Path $TestDrive -ChildPath $KeePassDatabaseFileName
+                Copy-Item -Path "$($PSScriptRoot)/$($KeePassDatabaseFileName)" -Destination $VaultPath
+
+                $RegisterSecretVaultPathOnlyParams = @{
+                    Name            = $VaultName
+                    ModuleName      = $ModulePath
+                    PassThru        = $true
+                    VaultParameters = @{
+                        Path = $VaultPath
+                    }
+                }
+                Microsoft.PowerShell.SecretManagement\Register-SecretVault @RegisterSecretVaultPathOnlyParams | Out-Null
+
+                Mock -Verifiable -CommandName 'Get-Credential' -MockWith { $VaultMasterKey }
+            }
+            AfterAll {
+                try {
+                    Microsoft.PowerShell.SecretManagement\Get-SecretVault -Name $VaultName -ErrorAction SilentlyContinue | Microsoft.PowerShell.SecretManagement\Unregister-SecretVault -ErrorAction SilentlyContinue
+                } catch [system.Exception] { }
+            }
+            It "should not have a variable 'Vault_$($VaultName)'" {
+                { (Get-Variable -Name "Vault_$VaultName" -Scope Script).Value } | Should -Throw
+            }
+            It "should throw on the first pass" {
+                { Test-SecretVault -VaultName $VaultName } | should -Throw
```

-Throw should be relatively explicit to the expected exception, in this case: | Should -Throw '*The composite key is invalid!*'